src/components/Resource.tsx

```diff
@@ -0,0 +1,307 @@
+      <h2 className="resource-title">💡 {resourceData.title}</h2>
+      <p className="resource-post-date">
+        Posted: {dateFormatting(resourceData.post_date)}
+      </p>
+      <p
+        onMouseEnter={() => setIsHovered(true)}
+        onMouseLeave={() => setIsHovered(false)}
+      >
+        {evaluateUsage(resourceData.usage)}
+      </p>
+      <div>
+        {isHovered && (
+          <div className="ctn-resource-usage-key">
+            <p> Used and recommended = 🌟</p>
+            <p> Not used but recommended = 🔎</p>
+            <p>Not recommended = 💩</p>
+          </div>
+        )}
+      </div>
+      <h4 className="resource-type">{resourceData.type}</h4>
+      <>
+        <h5 className="resource-description">{resourceData.description}</h5>
+        <div className="ctn-resource-tags">
+          {resourceData.tags.map((tag) => (
+            <span className="resource-tag" key={tag}>
+              {tag}
+            </span>
+          ))}
+        </div>
+      </>
+
+      <a className="resource-link-btn" href={resourceData.link}>
+        🔗Link to resource
+      </a>
+
+      {signedInUser !== undefined &&
+        !findResourceInLikes(resourceData.id, userLikes) && (
+          <div>
+            <button
+              className="false-like-resource-btn"
+              onClick={handleLikeClick}
+            >
+              👍|{resourceData.likes}
+            </button>
+            <button
+              className="false-dislike-resource-btn"
+              onClick={handleDislikeClick}
+            >
+              👎|{resourceData.dislikes}
+            </button>
+          </div>
+        )}
+      {signedInUser !== undefined &&
+        findResourceInLikes(resourceData.id, userLikes) === "like" && (
+          <div>
+            <button
+              className={
+                findResourceInLikes(resourceData.id, userLikes) +
+                "d-resource-btn"
+              }
+              onClick={handleRemoveLike}
+            >
+              👍|{resourceData.likes}
+            </button>
+            <button
+              className="false-dislike-resource-btn"
+              // onClick={handleDislikeClick}
+              disabled
+            >
+              👎|{resourceData.dislikes}
+            </button>
+          </div>
+        )}
+      {signedInUser !== undefined &&
+        findResourceInLikes(resourceData.id, userLikes) === "dislike" && (
+          <div>
+            <button
+              className="false-like-resource-btn"
+              // onClick={handleLikeClick}
+              disabled
+            >
+              👍|{resourceData.likes}
+            </button>
+            <button
+              className={
+                findResourceInLikes(resourceData.id, userLikes) +
+                "d-resource-btn"
+              }
+              onClick={handleRemoveDislike}
+            >
+              👎|{resourceData.dislikes}
+            </button>
+          </div>
+        )}
+      <div>
```

Comment:
Lots of repeated code here, three likes and dislikes implementation, a component might make this clearer